packages/react/src/components/TwoColumnPanel/TwoColumnPanel.tsx

```diff
@@ -0,0 +1,244 @@
+import React, {
+  forwardRef,
+  useState,
+  cloneElement,
+  isValidElement,
+  useRef,
+  useLayoutEffect,
+  useEffect
+} from 'react';
+import { useId } from 'react-id-generator';
+import FocusTrap from 'focus-trap-react';
+import Icon from '../Icon';
+import Tooltip from '../Tooltip';
+import ClickOutsideListener from '../ClickOutsideListener';
+import { ColumnLeft, ColumnRight } from './';
+import classnames from 'classnames';
+
+interface TwoColumnPanelProps extends React.HTMLAttributes<HTMLDivElement> {
+  initialCollapsed?: boolean;
+  showCollapsedPanelLabel?: string;
+  hideCollapsedPanelLabel?: string;
+}
+
+const TwoColumnPanel = forwardRef<HTMLDivElement, TwoColumnPanelProps>(
+  (
+    {
+      className,
+      children,
+      initialCollapsed = false,
+      showCollapsedPanelLabel = 'Show Panel',
+      hideCollapsedPanelLabel = 'Hide Panel',
+      ...props
+    },
+    ref
+  ) => {
+    const [isCollapsed, setCollapsed] = useState(initialCollapsed);
```

Comment:
We don't have to do this now, but it might make for a more flexible component if the collapsed state could be managed by the parent component.